### PR TITLE
[#166] Implement set_inheritance operation for /collections endpoint

### PR DESCRIPTION
**Pull request by korydraughn**
All tests pass.

No OIDC tests were run for this.

### CODE REVIEW

File: endpoints/collections/src/main.cpp [hunk 1]
```diff
@@ -689,6 +690,95 @@ namespace
...
+						res,
+						http::status::bad_request,
+						json{{"irods_response", {{"status_code", NOT_A_COLLECTION}}}}.dump()));
+				}
+
+				const auto enable_iter = _args.find("enable");
+				if (enable_iter == std::end(_args)) {
+					log::error("{}: Missing [enable] parameter.", fn);
+					return _sess_ptr->send(irods::http::fail(res, http::status::bad_request));
+				}
```
Comment (korydraughn) [inline endpoints/collections/src/main.cpp:734]: Move this above the call to get_connection.

File: API.md [hunk 2]
```diff
@@ -259,6 +259,36 @@ If an HTTP status code of 200 is returned, the body of the response will contain
 
 If there was an error, expect an HTTP status code in either the 4XX or 5XX range.
 
+### set_inheritance
```
Comment (trel) [inline API.md:262]: can we 'see' the inheritance setting somewhere already?   in the stat information perhaps?

Comment (trel) [reply API.md:262]: oh, it's in the stat.   thanks tests!

File: endpoints/collections/src/main.cpp [hunk 3]
```diff
@@ -689,6 +690,95 @@ namespace
...
+					log::error("{}: {}", fn, e.what());
+					// clang-format off
+					res.body() = json{
+						{"irods_response", {
+							{"status_code", e.code().value()},
+							{"status_message", e.what()}
+						}}
+					}.dump();
+					// clang-format on
+				}
```
Comment (alanking) [inline endpoints/collections/src/main.cpp:758]: This looks functionally identical to the `filesystem_error` catch just below this (L760-L764). Could we catch the `filesystem_error`s in one spot, or differentiate the error messages in some way?

**Review by alanking — COMMENTED** [review]
Yep, seems good.

I know it's planned for some time in the future, but I will nevertheless suggest adding some tests for values other than the valid ones. :)

File: endpoints/collections/src/main.cpp (hunk 3 above)
Comment (korydraughn) [reply endpoints/collections/src/main.cpp:758]: That catch block is a copy-paste situation. I must have been tired or something to let that through lol.

Will remove it.

**Comment by korydraughn** [comment]
I was about to add a test for invalid values, but then ran into the question of whether values other than 0 (for disable) should be considered invalid.

We can of course check for 0 or 1 and report an error, and that's good. However, that does imply additional checking of inputs, which may not be necessary.

I can see this going either way. I'm going to open an issue for this since it applies to many operations.

I'm going to hold on adding tests for invalid values until more users appear and have opinions. Things can change drastically depending on what the community learns/says and I don't want to over commit to anything just yet.

File: endpoints/collections/src/main.cpp (hunk 3 above)
Comment (korydraughn) [reply endpoints/collections/src/main.cpp]: Done.

**Comment by korydraughn** [comment]
Created #213.

**Comment by korydraughn** [comment]
Any last comments before I squash/pound?

**Comment by trel** [comment]
squash em

**Comment by korydraughn** [comment]
Squashed. Not sure if you meant to pound it too.

**Comment by korydraughn** [comment]
Adding pound. All tests pass.

If something needs to be changed, we can address it in a separate PR.

**Comment by korydraughn** [comment]
#'d

**Pull request merged**